### PR TITLE
Date aliases

### PR DESCRIPTION
**Pull request by pezholio**
Allows things like `all` to get all metrics, as well as `this-week`, `this-month` etc

### CODE REVIEW

**Comment by pezholio** [comment]
Any other ideas for useful date ranges gratefully received!

File: features/aliases.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      Then the response status should be "200"
+      And the JSON response should have "$.count" with the text "3"
+      And the JSON response should have "$.values[0].value.health" with the text "0.33"
+      And the JSON response should have "$.values[0].time" with the text "2013-12-24T15:00:00.000+00:00"
+      And the JSON response should have "$.values[1].value.health" with the text "0.34"
+      And the JSON response should have "$.values[1].time" with the text "2013-12-25T15:00:00.000+00:00"
+      And the JSON response should have "$.values[2].value.health" with the text "0.33"
+      And the JSON response should have "$.values[2].time" with the text "2013-12-30T15:00:00.000+00:00"
+
+    Scenario: Using `latest` alias
```
Comment (Floppy) [inline features/aliases.feature]: Shouldn't the default behaviour be to return the latest anyway?

Comment (Floppy) [inline features/aliases.feature]: Shouldn't the default behaviour be to return the latest anyway?

**Comment by Floppy** [comment]
I was thinking about non-aligned month periods, but we already have support for ISO8601 date ranges, so we can do things like "the last month" with P1M for instance, so we don't need those. `this-year` seems like it might be handy though.

File: features/aliases.feature (hunk 1 above)
Comment (pezholio) [inline features/aliases.feature]: The default returns a range of one month at the moment, `latest` returns a single point, useful for numbers etc

**Comment by pezholio** [comment]
Good plan, I'll add `this-year`

**Comment by pezholio** [comment]
How do the ISO8601 periods work, do they need a date to work from? So something like `/metrics/metric/P1M/{todays datetime}`? Is there an ISO8601-y way to express 'now'?

**Comment by Floppy** [comment]
I think your example is right, looking at the tests. I was looking for exactly a standard way of saying `now` but didn't find one. Perhaps `latest` could be `now` and used elsewhere as well, like in these cases?

**Comment by pezholio** [comment]
Actually, we should probably use the existing `*` wildcard, but using a wildcard and a duration blows up at the mo

**Comment by Floppy** [comment]
I had a thought. Instead of `this-week` and `this-month`, would it be better to stick with what Ruby named it and use `beginning-of-week` and `beginning-of-month`? If it's a duration, maybe `since-beginning-of-week`? More explicit, "this week" is perhaps confusing to some people? Especially on a Sunday :)

**Comment by Floppy** [comment]
Also, by analogy, `today` or `since-midnight` might be nice.

**Comment by Floppy** [comment]
I can stop commenting any time you like :)

**Comment by pezholio** [comment]
No

**Comment by pezholio** [comment]
Keep

**Comment by pezholio** [comment]
it

**Comment by pezholio** [comment]
up

**Comment by pezholio** [comment]
please

**Comment by Floppy** [comment]
Jelly Flood

**Comment by Floppy** [comment]
Shiny 👍

**Pull request merged**